src/Symfony/Component/Intl/Transliterator/EmojiTransliterator.php

```diff
@@ -11,144 +11,149 @@
 
 namespace Symfony\Component\Intl\Transliterator;
 
-if (\PHP_VERSION_ID >= 80200) {
-    final class EmojiTransliterator extends \Transliterator
-    {
-        use EmojiTransliteratorTrait;
-
-        private const QUICK_CHECK = "\xA9\xAE\xE2\xE3\xF0";
-        private const REVERSEABLE_IDS = [
-            'emoji-github' => 'github-emoji',
-            'emoji-slack' => 'slack-emoji',
-            'github-emoji' => 'emoji-github',
-            'slack-emoji' => 'emoji-slack',
-        ];
-
-        public readonly string $id;
-    }
+if (!class_exists(\Transliterator::class)) {
+    throw new \LogicException(sprintf('You cannot use the "%s\EmojiTransliterator" class as the "intl" extension is not installed. See https://www.php.net/manual/en/book.intl.php', __NAMESPACE__));
 } else {
-    final class EmojiTransliterator extends \Transliterator
+    /**
+    * @internal
+    */
+    trait EmojiTransliteratorTrait
```

Comment:
Unfortunately, it's not possible! 
```
SS............S.S...............S...............SEPHP Fatal error:  Cannot declare trait Symfony\Component\Intl\Transliterator\EmojiTransliteratorTrait, because the name is already in use in /home/gregoire/dev/github.com/lyrixx/symfony/src/Symfony/Component/Intl/Transliterator/EmojiTransliterator.php on line 49
```

It's because of OPCache, if I disable it, it's OK, I got my error message!

---

Step to reproduce:

1. disable intl
  ```
  sudo vim /etc/php/8.1/mods-available/intl.ini
  ```
2. remove `@requires extension intl` in `src/Symfony/Component/Intl/Tests/Transliterator/EmojiTransliteratorTest.php`
3. ```
   ./phpunit src/Symfony/Component/Intl/Tests/
   ```